test/CollateralFactory.t.sol

```diff
@@ -122,43 +146,116 @@ contract CollateralFactoryTest is Test {
         GlobalSettlementLike(0xeCa7B1F7e98B4ff65002DA43206fD2bE2d3aD0A3);
     TaxCollectorLike public taxCollector =
         TaxCollectorLike(0x3578743e922941911638a01521947ef1a81E0878);
+    StabilityFeeTreasuryLike public stabilityFeeTreasury =
+        StabilityFeeTreasuryLike(0xB3c5866f6690AbD50536683994Cc949697a64cd0);
     address public pauseProxy = 0xaDEA80Db702690A80cd123BA7ddCF6F541884E4f;
 
     function setUp() public {
+        address osmFactory = address(new OSMFactory());
         address joinFactory = address(new JoinFactory());
         address auctionHouseFactory = address(new AuctionHouseFactory());
+        address keeperIncentivesFactory = address(
+            new KeeperIncentivesFactory()
+        );
 
         // mainnet
         factory = new CollateralFactory(
+            pauseProxy,
             address(safeEngine),
             address(liquidationEngine),
             address(oracleRelayer),
             address(globalSettlement),
             address(taxCollector),
+            address(stabilityFeeTreasury),
+            osmFactory,
             joinFactory,
-            auctionHouseFactory
+            auctionHouseFactory,
+            keeperIncentivesFactory
         );
 
         vm.etch(pauseProxy, GetCode.at(address(factory)));
         factory = CollateralFactory(pauseProxy);
     }
 
-    function testDeployCollateralType() public {
-        address osm = 0xCb320D54d99250fD8D463B0Fea6785e44e78ce86; // mainnet WSTETH
+    function testDeployCollateralSpecificContracts() public {
         address token = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0; // mainnet WSTETH
-
-        (address joinAddress, address auctionHouseAddress) = factory
-            .deployCollateralType(
+        address priceFeed = 0xd0B49e3d0313bb3Cad456B62A99A0BcB366FB6a2; // COLLATERAL_ORACLE_WSTETH
+        address coinOracle = 0x7F9E3bC7c858Bf830906a0BD81F184708cc42260; // SYSTEM_COIN_ORACLE
```

Comment:
Why not use the mainnet coin oracle, `0xC8078539f56ae8E0e3741BAa8F8Ed939D63976a8`?

Reply:
True, nice catch, I went with the one on the changelog, still shows the previous one fixed to 1 wad. I'll amend the changelog today. Tests won't fail if we change the address (it would just impact rewards value when calling it)